File: demo/src/java/ast/KoKoAtom.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.ast;
import java.io.*;


public class KoKoAtom<T> implements KoKoAst{
   private T value;
   public T getValue(){return this.value;}
   
   public KoKoAtom(T value){
      this.value = value;
   }
   public void genCode(PrintStream out){
      out.print(this.value + " ");
   }
   public KoKoValue eval(KoKoContext ctx){
	   throw new KoKoEvalException("KoKoAtom: eval not implemented");
	   
   }
   public String toString(){
	   return value.toString();
   }
}

File: KoKoslan/src/kotlin/ast/KoKoAtom.kt
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.ast
import java.io.*


open class KoKoAtom<T>(private var value : T) : KoKoAst {

   fun getValue() : T {return this.value}

   override fun genCode(Out : PrintStream) : Unit{
      Out.print(this.value + " ")
   }
   override fun eval(ctx : KoKoContext) : KoKoValue {
	   throw KoKoEvalException("KoKoAtom: eval not implemented") 
   }
   override fun toString() : String {
	   return value.toString()
   }
}